In [6]:
import pandas as pd
import numpy as np

### EXERCISE 1
95th percentile of earthquake magnitude in Japan using `mb` magnitude type

In [7]:
quake = pd.read_csv('./data/parsed.csv')
quake.tail(5)

alert  cdi        code  \
9327   NaN  NaN    73086771   
9328   NaN  NaN    38063967   
9329   NaN  NaN  2018261000   
9330   NaN  NaN    38063959   
9331   NaN  NaN    38063935   

                                                 detail     dmin  felt    gap  \
9327  https://earthquake.usgs.gov/fdsnws/event/1/que...  0.01806   NaN  185.0   
9328  https://earthquake.usgs.gov/fdsnws/event/1/que...  0.03041   NaN   50.0   
9329  https://earthquake.usgs.gov/fdsnws/event/1/que...  0.45260   NaN  276.0   
9330  https://earthquake.usgs.gov/fdsnws/event/1/que...  0.01865   NaN   61.0   
9331  https://earthquake.usgs.gov/fdsnws/event/1/que...  0.01698   NaN   39.0   

                 ids   mag magType  ...    status           time  \
9327    ,nc73086771,  0.62      md  ...  reviewed  1537230228060   
9328    ,ci38063967,  1.00      ml  ...  reviewed  1537230135130   
9329  ,pr2018261000,  2.40      md  ...  reviewed  1537229908180   
9330    ,ci38063959,  1.10      ml  ...  reviewed  1537229545350   
9331    ,ci38063935,  0.66      ml  ...  reviewed  1537228864470   

                                         title tsunami        type  \
9327      M 0.6 - 9km ENE of Mammoth Lakes, CA       0  earthquake   
9328               M 1.0 - 3km W of Julian, CA       0  earthquake   
9329  M 2.4 - 35km NNE of Hatillo, Puerto Rico       0  earthquake   
9330             M 1.1 - 9km NE of Aguanga, CA       0  earthquake   
9331             M 0.7 - 9km NE of Aguanga, CA       0  earthquake   

                                                  types     tz        updated  \
9327         ,geoserve,nearby-cities,origin,phase-data, -480.0  1537285598315   
9328  ,geoserve,nearby-cities,origin,phase-data,scit... -480.0  1537276800970   
9329                       ,geoserve,origin,phase-data, -240.0  1537243777410   
9330  ,focal-mechanism,geoserve,nearby-cities,origin... -480.0  1537230211640   
9331  ,focal-mechanism,geoserve,nearby-cities,origin... -480.0  1537305830770   

                                                    url parsed_place  
9327  https://earthquake.usgs.gov/earthquakes/eventp...   California  
9328  https://earthquake.usgs.gov/earthquakes/eventp...   California  
9329  https://earthquake.usgs.gov/earthquakes/eventp...  Puerto Rico  
9330  https://earthquake.usgs.gov/earthquakes/eventp...   California  
9331  https://earthquake.usgs.gov/earthquakes/eventp...   California  

[5 rows x 27 columns]

In [24]:
quake.tsunami.max()

1

In [17]:
ex1 = quake.loc[(quake.magType == 'mb') & (quake.place.str.contains(', Japan'))]
ex1.head()

alert  cdi      code                                             detail  \
67     NaN  NaN  1000hbqa  https://earthquake.usgs.gov/fdsnws/event/1/que...   
713    NaN  NaN  1000hah8  https://earthquake.usgs.gov/fdsnws/event/1/que...   
1124   NaN  NaN  1000h9la  https://earthquake.usgs.gov/fdsnws/event/1/que...   
1309   NaN  NaN  1000h952  https://earthquake.usgs.gov/fdsnws/event/1/que...   
1398   NaN  NaN  1000h8zx  https://earthquake.usgs.gov/fdsnws/event/1/que...   

       dmin  felt    gap           ids  mag magType  ...    status  \
67    1.415   NaN   72.0  ,us1000hbqa,  4.6      mb  ...  reviewed   
713   1.141   NaN   82.0  ,us1000hah8,  4.7      mb  ...  reviewed   
1124  1.737   NaN  135.0  ,us1000h9la,  4.6      mb  ...  reviewed   
1309  0.897   NaN  136.0  ,us1000h952,  4.5      mb  ...  reviewed   
1398  0.555   NaN  124.0  ,us1000h8zx,  4.4      mb  ...  reviewed   

               time                                    title tsunami  \
67    1539448501800         M 4.6 - 160km NNW of Nago, Japan       0   
713   1539238726290         M 4.7 - 139km WSW of Naze, Japan       0   
1124  1539115120470      M 4.6 - 53km ESE of Kamaishi, Japan       0   
1309  1539058606580        M 4.5 - 80km ENE of Misawa, Japan       0   
1398  1539034912990  M 4.4 - 65km NNE of Hachijo-jima, Japan       0   

            type                         types     tz        updated  \
67    earthquake  ,geoserve,origin,phase-data,  480.0  1539449501040   
713   earthquake  ,geoserve,origin,phase-data,  540.0  1539240344040   
1124  earthquake  ,geoserve,origin,phase-data,  540.0  1539119067040   
1309  earthquake  ,geoserve,origin,phase-data,  540.0  1539060436040   
1398  earthquake  ,geoserve,origin,phase-data,  540.0  1539062940040   

                                                    url parsed_place  
67    https://earthquake.usgs.gov/earthquakes/eventp...        Japan  
713   https://earthquake.usgs.gov/earthquakes/eventp...        Japan  
1124  https://earthquake.usgs.gov/earthquakes/eventp...        Japan  
1309  https://earthquake.usgs.gov/earthquakes/eventp...        Japan  
1398  https://earthquake.usgs.gov/earthquakes/eventp...        Japan  

[5 rows x 27 columns]

In [18]:
ex1.shape

(50, 27)

In [20]:
ex1.describe(percentiles=[0.05, 0.95])

cdi       dmin       felt         gap        mag  mmi  nst  \
count  16.000000  50.000000  16.000000   50.000000  50.000000  0.0  0.0   
mean    2.881250   1.485080  11.187500  108.880000   4.530000  NaN  NaN   
std     0.883341   0.802257  20.672748   31.676709   0.266688  NaN  NaN   
min     1.000000   0.223000   1.000000   33.000000   4.000000  NaN  NaN   
5%      1.000000   0.533550   1.000000   68.700000   4.100000  NaN  NaN   
50%     3.100000   1.271000   3.500000  112.000000   4.600000  NaN  NaN   
95%     3.875000   2.583200  35.500000  155.650000   4.900000  NaN  NaN   
max     4.100000   4.830000  85.000000  167.000000   5.400000  NaN  NaN   

             rms         sig          time  tsunami          tz       updated  
count  50.000000   50.000000  5.000000e+01     50.0   50.000000  5.000000e+01  
mean    0.801800  318.080000  1.538218e+12      0.0  547.200000  1.538746e+12  
std     0.258589   38.570286  6.404535e+08      0.0   26.111222  5.518279e+08  
min     0.270000  246.000000  1.537258e+12      0.0  480.000000  1.537430e+12  
5%      0.428000  259.000000  1.537409e+12      0.0  540.000000  1.537614e+12  
50%     0.810000  326.000000  1.538268e+12      0.0  540.000000  1.538886e+12  
95%     1.335500  371.750000  1.539090e+12      0.0  600.000000  1.539428e+12  
max     1.400000  451.000000  1.539449e+12      0.0  600.000000  1.539533e+12

#### EXERCISE 1 ANSWER
95% of Japan with `mb` magtype = 4.9

### EXERCISE 2
%age of earthquakes in Indonesia coupled with tsunamis

In [40]:
pctInTsu = quake.loc[(quake.parsed_place == 'Indonesia') & (quake.tsunami == 1)].shape[0] / quake.loc[(quake.parsed_place == 'Indonesia')].shape[0] * 100
round(pctInTsu, 2)

23.13

In [48]:
f"{quake[quake.parsed_place == 'Indonesia'].tsunami.value_counts(normalize=True).iloc[1,]:.2%}"
quake[quake.parsed_place == 'Indonesia'].tsunami.value_counts()
# so value_counts with (normalize=True) gives the percentage proportions of each value in the column
# chaining allows sequential filtering ... Indonesia'].tsunami... etc.

0    113
1     34
Name: tsunami, dtype: int64

#### EXERCISE 2 ANSWER
23.13%

### EXERCISE 3
Summary statistics for earthquakes in Nevada

In [50]:
quake[(quake.parsed_place == 'Nevada')].describe()

cdi        dmin       felt         gap         mag   mmi  \
count  15.000000  681.000000  15.000000  681.000000  681.000000  1.00   
mean    2.440000    0.166199   2.400000  153.668120    0.500073  2.84   
std     0.501142    0.166228   4.626013   68.735302    0.696710   NaN   
min     2.000000    0.001000   1.000000   29.140000   -0.500000  2.84   
25%     2.000000    0.053000   1.000000   97.380000   -0.100000  2.84   
50%     2.200000    0.112000   1.000000  149.140000    0.400000  2.84   
75%     2.900000    0.233000   1.000000  199.720000    0.900000  2.84   
max     3.300000    1.414000  19.000000  355.910000    2.900000  2.84   

              nst         rms         sig          time  tsunami     tz  \
count  681.000000  681.000000  681.000000  6.810000e+02    681.0  681.0   
mean    12.618209    0.151986   10.970631  1.538314e+12      0.0 -480.0   
std      9.866963    0.084662   19.607150  5.965637e+08      0.0    0.0   
min      3.000000    0.000500    0.000000  1.537247e+12      0.0 -480.0   
25%      6.000000    0.106900    0.000000  1.537854e+12      0.0 -480.0   
50%     10.000000    0.146300    2.000000  1.538280e+12      0.0 -480.0   
75%     16.000000    0.187100   12.000000  1.538821e+12      0.0 -480.0   
max     61.000000    0.863400  129.000000  1.539461e+12      0.0 -480.0   

            updated  
count  6.810000e+02  
mean   1.538402e+12  
std    6.010951e+08  
min    1.537307e+12  
25%    1.537928e+12  
50%    1.538428e+12  
75%    1.538878e+12  
max    1.539483e+12

#### EXERCISE 3 DONE

### EXCERCISE 4
Add column indicating is earthquake happened in country or US state that is on Ring of Fire

In [51]:
quake.parsed_place.sort_values().unique()

array(['Afghanistan', 'Alaska', 'Argentina', 'Arizona', 'Arkansas',
       'Ascension Island', 'Australia', 'Azerbaijan', 'Balleny Islands',
       'Barbuda', 'Bolivia', 'British Virgin Islands', 'Burma',
       'California', 'Canada', 'Carlsberg Ridge',
       'Central East Pacific Rise', 'Central Mid-Atlantic Ridge', 'Chile',
       'China', 'Christmas Island', 'Colombia', 'Colorado', 'Costa Rica',
       'Dominican Republic', 'East Timor', 'Ecuador', 'El Salvador',
       'Fiji', 'Greece', 'Greenland', 'Guam', 'Guatemala', 'Haiti',
       'Hawaii', 'Honduras', 'Idaho', 'Illinois', 'India',
       'Indian Ocean Triple Junction', 'Indonesia', 'Iran', 'Iraq',
       'Italy', 'Jamaica', 'Japan', 'Kansas', 'Kentucky',
       'Kermadec Islands', 'Kuril Islands', 'Kyrgyzstan', 'Martinique',
       'Mauritius', 'Mayotte', 'Mexico', 'Mid-Indian Ridge', 'Missouri',
       'Montana', 'Nevada', 'New Caledonia', 'New Hampshire',
       'New Mexico', 'New Zealand', 'Nicaragua', 'North Carolina',


In [69]:
ex4 = quake.loc[(quake.parsed_place.isin(['Alaska', 'Antarctic', 'Bolivia', 'California', 'Canada', 'Chile', 'Costa Rica', 'Ecuador', 'Fiji', 'Guatemala', 'Indonesia', 'Japan', 'Kermadec Islands', 'Mexico', 'New Zealand', 'Peru', 'Phillipines', 'Russia', 'Taiwan', 'Tonga', 'Washington']))]

In [81]:
quake.parsed_place.unique()

array(['California', 'Dominican Republic', 'Alaska', 'Indonesia',
       'Canada', 'Puerto Rico', 'Montana', 'Nevada', 'Christmas Island',
       'Hawaii', 'Northern Mariana Islands', 'Japan', 'Ecuador',
       'Vanuatu', 'Mexico', 'Russia', 'British Virgin Islands',
       'Washington', 'Papua New Guinea', 'Fiji', 'U.S. Virgin Islands',
       'Chile', 'Peru', 'Yemen', 'Guatemala', 'Kansas', 'Australia',
       'Wyoming', 'Kuril Islands', 'Oklahoma', 'Tennessee',
       'Pacific-Antarctic Ridge', 'Utah', 'Colombia', 'Argentina',
       'Oregon', 'Greece', 'Missouri', 'Tajikistan',
       'Northern Mid-Atlantic Ridge', 'Sumatra', 'Solomon Islands',
       'Burma', 'Taiwan', 'Nicaragua',
       'South Georgia and South Sandwich Islands', 'Idaho', 'Kyrgyzstan',
       'Arizona', 'Tonga', 'Northern East Pacific Rise', 'South Africa',
       'Southern Mid-Atlantic Ridge', 'Costa Rica', 'China',
       'Philippines', 'Haiti', 'Jamaica', 'Kentucky', 'New Zealand',
       'Iran', 'Afghanistan

In [91]:
quake['rof'] = quake.parsed_place.str.contains(r'|'.join(['Alaska', 'Antarctic', 'Bolivia', 'California', 'Canada', 'Chile', 'Costa Rica', 'Ecuador', 'Fiji', 'Guatemala', 'Indonesia', 'Japan', 'Kermadec Islands', '^Mexico', 'New Zealand', 'Peru', 'Philippines', 'Russia', 'Taiwan', 'Tonga', 'Washington']))

In [92]:
quake['ring_of_fire'] = quake.parsed_place.str.contains(r'|'.join([
    'Alaska', 'Antarctic', 'Bolivia', 'California', 'Canada',
    'Chile', 'Costa Rica', 'Ecuador', 'Fiji', 'Guatemala',
    'Indonesia', 'Japan', 'Kermadec Islands', '^Mexico',
    'New Zealand', 'Peru', 'Philippines', 'Russia',
    'Taiwan', 'Tonga', 'Washington' 
]))
quake.rof.unique()

array([ True, False])

In [93]:
quake[(quake.rof != quake.ring_of_fire)]

Empty DataFrame
Columns: [alert, cdi, code, detail, dmin, felt, gap, ids, mag, magType, mmi, net, nst, place, rms, sig, sources, status, time, title, tsunami, type, types, tz, updated, url, parsed_place, rof, ring_of_fire]
Index: []

[0 rows x 29 columns]

#### EXERCISE 4 DONE

### EXERCISE 5
Number of quakes in RoF locations and number outside

In [94]:
quake.rof.value_counts()

True     7188
False    2144
Name: rof, dtype: int64

In [104]:
quake.loc[quake.rof, 'tsunami'].describe()

count    7188.00000
mean        0.00626
std         0.07888
min         0.00000
25%         0.00000
50%         0.00000
75%         0.00000
max         1.00000
Name: tsunami, dtype: float64

In [106]:
quake.shape

(9332, 29)